### DISCUSSION TITLE
Is there any easier method on the horizon?

### DISCUSSION THREAD

**Discussion by pixelwave** [alandtse/tesla#1014] [General]
Is there any easier method on the horizon? I was quite happy with the Tesla Integration so far that just required to generate a token with my phone app.

Now all this exposing HA to the internet, SSL certificates, (reverse) proxy setup … WHY?

:pensive:

**Comment by skipishere** [comment]
Blame Tesla, we'd love for it to be simpler but this is a limitation of the way they have implemented the fleet API

**Comment by nickt444** [comment]
Agreed. You dont need to expose HA to the internet - you just need a DNS domain, SSL cert and host the public key generated on HA. I just registered a cheap domain and use Azure Static Web apps as a low-cost hosting option for the key (it gives you the matching SSL cert for free). PITA compared to the old API, but it works.
Next headache is the 50/day command limit. Considering moving to BLE as I keep randomly getting throttled...

**Reply by madsdyd** [reply]
I was not aware of this limit. I used to change my amps on my solar charching every 30 seconds ... this will definitively be more than 50 during a solar charge!

Can I read more about this limit somewhere? What happens when the limit is enforced? What is the throttling? How would you move to BLE? That sounds quite interessting!

**Reply by nickt444** [reply]
limits are [here](https://developer.tesla.com/docs/fleet-api#membership-levels)
if you turn on verbose logging on the proxy component, you will see http 429 response codes when you hit the limit. Total PITA.
50/day is insanely low, but I guess the price is right.

**Comment by jesperholstjh** [comment]
No, there isn't. This is 100% Teslas fault, nothing the author can do about the changes they made.

**Comment by darek-margas** [comment]
After introducing new API, Tesla provided band-aid solution in a form of a proxy which translates old style calls into new API commands. As it is in Go, I believe it won't be possible to reuse code here. However, nothing is stopping us from writing new integration in Python which natively calls new API. Just someone has to start new project and keep writing. I don't hold my breath, however. Seeing as all commercial implementation are stuck with Tesla's proxy, even as paid project no-one want's to take on it. We need to see someone talented and with lots of courage and spare time to start it. It would be more likely if proxy was not provided.

**Reply by skipishere** [reply]
The proxy that @llamafilm has provided is a python version of the Tesla proxy and is the only way to control the car when it is out and about.

The other solution is the BLE (Bluetooth low energy) way which is fine for at home and a few people have been trying it out to get around the fleet APIs seriously low request limits.

**Reply by madsdyd** [reply]
Hi. Where can I read more about the BLE approach? That sounds very interessting!

**Reply by jesperholstjh** [reply]
https://github.com/tesla-local-control/tesla-local-control-addon

Works really well.
